lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -1153,7 +1153,17 @@ private function Constant()
             throw AnnotationException::semanticalErrorConstants($identifier, $this->context);
         }
 
-        return constant($identifier);
+		$value = constant($identifier);
+
+		/** @noinspection PhpElementIsNotAvailableInCurrentPhpVersionInspection */
+		if (interface_exists('\UnitEnum') && $value instanceof \UnitEnum) {
+			if (!property_exists($value, 'value')) {
+				throw AnnotationException::semanticalErrorConstants('Enum ' . $identifier. ' is not backed enum, only backed enums can be used!');
+			}
+			$value = $value->value;
```

Comment:
I don't think turning `MyEnum::FOO` into `MyEnum::FOO->value` is the right way to support enums in annotations. To me, that's quite confusing.

Reply:
This way when someone changes class with constants used in annotations to backed enum with cases, no additional changes in annotations are required. What would be alternative ?

Reply:
well, if `MyEnum::FOO` is used in an annotation, I would expect it to actually mean `MyEnum::FOO` and not a weird behavior of making it mean `MyEnum::FOO->value`

Reply:
Btw, I'm quite sure `MyEnum::FOO` actually works fine today and that your special logic actually breaks such usage.

Reply:
For example when using backed enum in DiscripimatorMap (which i do think should work/be supported)

/\*\* 
 \* @ORM\DiscriminatorMap({
 \*     MyEnum::FOO=Example::class
 \*	 })
 \* /

Annotations fail with message `Illegal offset type`  because the enum case (and not the enum case value) is used as array key in function Arrayx in DocParser (DocParser:1357). 
And using MyEnum::FOO->value directly in the annotation does not work either.
I haven't found a proper way (other than copying the value a second time as constant instead of case) to use backed enum in annotation for discriminator map